aopy/analysis.py

```diff
@@ -1374,6 +1374,45 @@ def match_selectivity_accLLR(test_data_altcond, train_data_altcond, train_data_n
 
         return noisy_test_data
 
+def calc_task_rel_dims(neural_data, kin_data):
+    '''
+    Calculates the task relevant dimensions by regressing neural activity against kinematic data using least squares.
+    If the input neural data is 3D, all trials will be concatenated to calculate the subspace. 
+    Calculation is based on the approach used in Sun et al. 2022 https://doi.org/10.1038/s41586-021-04329-x
+
+
+    Args:
+        neural_data (list of (nt, nch) or (nt, nch)): Input neural data to regress against kinematic activity
+        kin_data (nt, ndim): Kinematic variables, commonly position or instantaneous velocity. 'ndims' refers to the number of physical dimensions that define the kinematic data (i.e. X and Y)
+
+    Returns:
+        (nch, ndim): Subspace that best predicts kinematic variables.
+        (list )
+
+    '''
+    # If a list of segments from trials, concatenate them into one larget timeseries
+    if type(neural_data) == list:
+        ntrials = len(neural_data)
+        conc_neural_data = np.vstack(neural_data)
+        conc_kin_data = np.vstack(kin_data)
+    else:
+        neural_data = [neural_data]
+        ntrials = 1
+
+    # Calculate task relevant subspace 
+    task_subspace = np.linalg.pinv(conc_kin_data.T @ conc_kin_data) @ conc_kin_data.T @ conc_neural_data
+
+    # Project neural data onto task subspace
+    projected_data = []
+    
+    for itrial in range(ntrials):
+        projected_data.append(neural_data[itrial] @ np.linalg.pinv(task_subspace))
```

Comment:
For generalization, it might be good to add an option to choose if projected neural data is concatenated or not?